klient/k8s/resources/resources.go

```diff
@@ -121,3 +133,49 @@ func (r *Resources) List(ctx context.Context, objs k8s.ObjectList, opts ...ListO
 	o := &cr.ListOptions{Raw: listOptions}
 	return r.client.List(ctx, objs, o)
 }
+
+func WithLabelSelector(sel string) ListOption {
+	return func(lo *metav1.ListOptions) { lo.LabelSelector = sel }
+}
+
+func WithFieldSelector(sel string) ListOption {
+	return func(lo *metav1.ListOptions) { lo.FieldSelector = sel }
+}
+
+func WithTimeout(to time.Duration) ListOption {
+	t := to.Milliseconds()
+	return func(lo *metav1.ListOptions) { lo.TimeoutSeconds = &t }
+}
+
+// PatchOption is used to provide additional arguments to the Patch call.
+type PatchOption func(*metav1.PatchOptions)
+
+// Patch patches portion of object `orig` with data from object `patch`
+func (r *Resources) Patch(ctx context.Context, objs k8s.Object, patch k8s.Patch, opts ...PatchOption) error {
+	patchOptions := &metav1.PatchOptions{}
+
+	for _, fn := range opts {
+		fn(patchOptions)
+	}
+
+	p := cr.RawPatch(patch.PatchType, patch.Data)
+
+	o := &cr.PatchOptions{Raw: patchOptions}
+	return r.client.Patch(ctx, objs, p, o)
+}
+
+// BuildFromJSON encodes a JSON string as a typed (or unstructured) object representation.
+// TODO: needs to define the object structure to full fill the functionality
+func (r *Resources) BuildFromJSON(obj k8s.Object, jsonData string) error {
+	return json.Unmarshal([]byte(jsonData), &obj)
```

Comment:
I don't think we should use the plain JSON encoder.
Instead, look in the [api-machinery runtime for tests on how to use encoders](https://sourcegraph.com/github.com/kubernetes/apimachinery@master/-/blob/pkg/runtime/embedded_test.go)

Reply:
Actually this package is better and it applies the proper object conversions https://sourcegraph.com/github.com/kubernetes/apimachinery@master/-/tree/pkg/util/json

Reply:
Just marked this PR to WIP to update the changes for JSON encoding .

Reply:
JSON encoding will be taken care as part of #43